### PR TITLE
Reduced queries count to RDS while building media tree

### PR DESCRIPTION
**Pull request by boajer**
Measures to improve https://oat-sa.atlassian.net/browse/CON-111

Sample request `http://depp-authoring.docker.localhost/taoItems/ItemContent/files?path=taomedia%3A%2F%2Fmediamanager%2F&uri=https%3A%2F%2Fdepp-auth.taocloud.org%2Ftao.rdf%23i16049178955445394934&lang=en-US&filters=video%2Fmp4%2Cvideo%2Favi%2Cvideo%2Fogv%2Cvideo%2Fmpeg%2Cvideo%2Fogg%2Cvideo%2Fquicktime%2Cvideo%2Fwebm%2Cvideo%2Fx-ms-wmv%2Cvideo%2Fx-flv%2Caudio%2Fmp3%2Caudio%2Fvnd.wav%2Caudio%2Fogg%2Caudio%2Fvorbis%2Caudio%2Fwebm%2Caudio%2Fmpeg%2Capplication%2Fogg%2Caudio%2Faac%2Caudio%2Fwav%2Caudio%2Fflac` 

Tested  with `5372` media files in library
before change  - `26711` queries 
after change - `2952` queries

### CODE REVIEW

File: model/MediaSource.php [hunk 1]
```diff
@@ -137,34 +136,38 @@ public function getDirectory($parentLink = '', $acceptableMime = array(), $depth
...
+            throw new \tao_models_classes_FileNotFoundException($link);
+        }
 
         // add the alt text to file array
-        $altArray = $resource->getPropertyValues($this->getProperty(MediaService::PROPERTY_ALT_TEXT));
+        $altArray = $propertiesValues[MediaService::PROPERTY_ALT_TEXT] ?? null;
         $alt = $resource->getLabel();
         if (count($altArray) > 0) {
-            $alt = $altArray[0];
+            $alt = (string)$altArray[0];
```
Comment (gabrielfs7) [inline model/MediaSource.php:160]: SO we can have more than one alt text. Did not know it was possible. It was there before, I am just surprised.

File: model/MediaSource.php [hunk 2]
```diff
@@ -137,34 +136,38 @@ public function getDirectory($parentLink = '', $acceptableMime = array(), $depth
...
 
-        $fileLink = $resource->getUniquePropertyValue($this->getProperty(MediaService::PROPERTY_LINK));
+        $fileLink = $propertiesValues[MediaService::PROPERTY_LINK][0] ?? null;
+        $mime = $propertiesValues[MediaService::PROPERTY_MIME_TYPE][0] ?? null;
         $fileLink = $fileLink instanceof \core_kernel_classes_Resource ? $fileLink->getUri() : (string)$fileLink;
-        $file = null;
-        $mime = (string) $resource->getUniquePropertyValue($this->getProperty(MediaService::PROPERTY_MIME_TYPE));
+
+        if (!isset($mime, $fileLink)){
+            throw new \tao_models_classes_FileNotFoundException($link);
```
Comment (gabrielfs7) [inline model/MediaSource.php:153]: Is this new exception here safe to be triggered? I know there is a try/catch on `getDirectory` method, but will any other place break with this?

**Review by gabrielfs7 — COMMENTED** [review]
Good job! Great performance difference! :) 

Some questions:

- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/MediaSource.php (hunk 1 above)
Comment (boajer) [reply model/MediaSource.php:160]: That is historical behavior, surprising me as well, but I decided not change this

File: model/MediaSource.php (hunk 2 above)
Comment (boajer) [reply model/MediaSource.php:153]: That has been thrown before at https://github.com/oat-sa/extension-tao-mediamanager/pull/239/files/31e226eeb8f3ee7ec60d57de4765cc77d6bdcfb6#diff-d411d51cd4406db872695a30a830894330f127153df3ae03be7a1caf314abe10L143

File: model/MediaSource.php (hunk 1 above)
Comment (gabrielfs7) [reply model/MediaSource.php:160]: thanks for the clarification :)

File: model/MediaSource.php (hunk 2 above)
Comment (gabrielfs7) [reply model/MediaSource.php:153]: Thanks for the clarification! :) 

I saw `$resource->exists()` was counting all the resource types. Is there a chance this verification removal will cause another bug? Seems this is the standard way by some reason. I saw @jbout authored that code, maybe we could ask him to double check.

File: model/MediaSource.php [hunk 3]
```diff
@@ -137,34 +136,38 @@ public function getDirectory($parentLink = '', $acceptableMime = array(), $depth
      */
     public function getFileInfo($link)
     {
-        // get the media link from the resource
         $resource = $this->getResource(\tao_helpers_Uri::decode($link));
-        if (!$resource->exists()) {
-            throw new \tao_models_classes_FileNotFoundException($link);
-        }
+        $properties = [
```
Comment (gabrielfs7) [inline model/MediaSource.php:140]: Was it not required to fix `\oat\taoMediaManager\test\integration\model\MediaSourceTest`?

Comment (boajer) [reply model/MediaSource.php:140]: Tests are still green for the altered methods.

**Review by gabrielfs7 — APPROVED** [review]
Code LGTM. Good improvement. I would suggest just check with @jbout if removing the resource->exists is safe or we miss other validation

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
lgtm, 
tested on pre-prod
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by shpran — APPROVED** [review]
1 minor comment, can be skipped.
Good job!

- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by bartlomiejmarszal — APPROVED** [review]
I do not recommend to commit anything else than the actual fix in this hotfix PR. Code looks fine.

**Pull request merged**